synapse/handlers/presence.py

```diff
@@ -1513,8 +1517,8 @@ async def get_new_events(
             presence_updates = list(users_to_state.values())
 
         # Remove the user from the list of users to receive all presence
-        if user_id in self.get_module_api()._send_full_presence_to_local_users:
-            self.get_module_api()._send_full_presence_to_local_users.remove(user_id)
+        if user_in_users_to_send_full_presence_to:
+            await self.store.remove_user_from_users_to_send_full_presence_to(user_id)
```

Comment:
I don't think this works from the perspective of multiple devices (or if they do a `/sync`, which they fail to read the response of, and so have to repeat their `/sync` request). I think we can fix that by added a `stream_id` column to the table and only returning the full presence if the stream_id in the sync token is less than that one?

Reply:
Good point. In that case we wouldn't want to remove rows from the table once a device /sync's successfully.

Reply:
To be clear: I assume the value of `stream_id` in `users_to_send_full_presence_to` would be based on the presence stream ID that's already included in the sync token, rather than adding a stream ID to the token.